sherpa/astro/io/crates_backend.py

```diff
@@ -1409,112 +1430,78 @@ def mkcol(name, vals, units=None):
 
     ds.write(filename, clobber=True)
 
-    # Add the TLMIN value for the F_CHAN column
-    ds = cxcdm.dmDatasetOpen(filename, update=True)
-    bl = cxcdm.dmBlockOpen("MATRIX", ds, update=True)
-    col = cxcdm.dmTableOpenColumn(bl, "F_CHAN")
-
-    # The lo and hi vals must have the same type, so force them to be
-    # int (so we do not have to worry about whether this should be
-    # Int16 or Int32). The OFFSET value should be an int, but let's
-    # convert it too to make sure.
-    #
-    lval, hval = cxcdm.dmDescriptorGetRange(col)
-    cxcdm.dmDescriptorSetRange(col, int(matrix_data["OFFSET"]), int(hval))
-    cxcdm.dmBlockClose(bl)
-    cxcdm.dmDatasetClose(ds)
-
 
 def set_arrays(filename, args, fields=None, ascii=True, clobber=False):
 
+    # Historically the clobber command has been checked before
+    # processing the data, so do so here.
+    #
     check_clobber(filename, clobber)
 
-    if not numpy.iterable(args) or len(args) == 0:
-        raise IOErr('noarrayswrite')
-
-    if not numpy.iterable(args[0]):
-        raise IOErr('noarrayswrite')
+    # Check args is a sequence of sequences (although not a complete
+    # check).
+    #
+    try:
+        size = len(args[0])
+    except (TypeError, IndexError) as exc:
+        raise IOErr('noarrayswrite') from exc
 
```

Comment:
An alternative would be:
```suggestion
    if not isinstance(args[0], (collections.abs.Sequence, np.ndarray))
        raise IOErr('noarrayswrite') from exc

```
It's slightly different in that is checks the class hirachy, not duck-typing, but in practice, it's probably the same. I don't think either is "better", but my version is shorter ;-)

Same applies for the lines below.

Reply:
Your version doesn't check the case of `args` not being indexable, unless I'm mistaken.

Now, some of my code changes are written to try and improve the typing results, and so some of my thinking could be being swayed by that, so it's good to check on my thinking here.

Reply:
Are there any Sequences that are not indexable? Of course, we could have a multi-D numpy array etc., so it's not a complete check either...

Reply:
What happens if you send in `arg=None`?

Reply:
Then it will fail, because `None` is no subscriptable. So, let's keep your version. As I said above, I'm not sure testing the hierarchy (as I do) is any better anyway.